event_handling/src/Worker.cpp

```diff
@@ -1,76 +1,74 @@
 #include "essentials/Worker.h"
-#include "essentials/ITrigger.h"
-#include "essentials/Timer.h"
 
 #include <string>
+#include <thread>
 
 namespace essentials
 {
 
-Worker::Worker(std::string name)
-        : name(name)
-        , started(true)
-        , runCV()
+Worker::Worker(std::string name, std::chrono::milliseconds msInterval, std::chrono::milliseconds msDelayedStart)
+        : _name(name)
+        , _started(true)
+        , _runCV()
```

Comment:
well, now it looks like you can get rid of everything in worker class and just say 
using Worker = NotifyTimer;
and things shouldn't break.  Only thing is rather than run() method whatever passed in constructor would be called. If you still want to keep this class, at least don't spawn any thread here, use callback version of NotifyTimer and simply call run() method in timer callback

Reply:
gone